src/test/java/net/snowflake/client/jdbc/SnowflakeDriverLatestIT.java

```diff
@@ -69,6 +72,33 @@ public void testGetSessionID() throws Throwable {
     assertEquals(sessionID, rset.getString(1));
   }
 
+  @Test
+  public void testPutThreshold() throws SQLException {
+    try (Connection connection = getConnection()) {
+      // assert that threshold equals default 200 from server side
+      SFSession sfSession = connection.unwrap(SnowflakeConnectionV1.class).getSfSession();
+      Statement statement = connection.createStatement();
+      SFStatement sfStatement = statement.unwrap(SnowflakeStatementV1.class).getSfStatement();
+      statement.execute("CREATE OR REPLACE STAGE PUTTHRESHOLDSTAGE");
+      String command =
+          "PUT file://" + getFullPathFileInResource(TEST_DATA_FILE) + " @PUTTHRESHOLDSTAGE";
+      SnowflakeFileTransferAgent agent =
+          new SnowflakeFileTransferAgent(command, sfSession, sfStatement);
+      assertEquals(200 * 1024 * 1024, agent.getBigFileThreshold());
+      // assert that setting threshold via put statement directly sets the big file threshold
+      // appropriately
+      String commandWithPut = command + " threshold=314572800";
+      agent = new SnowflakeFileTransferAgent(commandWithPut, sfSession, sfStatement);
+      assertEquals(314572800, agent.getBigFileThreshold());
+      // assert that after put statement, threshold goes back to previous session threshold
+      agent = new SnowflakeFileTransferAgent(command, sfSession, sfStatement);
+      assertEquals(200 * 1024 * 1024, agent.getBigFileThreshold());
+      statement.close();
+    } catch (SQLException ex) {
+      throw ex;
+    }
```

Comment:
Maybe we can also add a test for `threshold=invalid number`, e.g. negative number.

Reply:
Good idea! Added that.